packages/xy-chart/src/series/BarSeries.jsx

```diff
@@ -46,24 +45,30 @@ export default class BarSeries extends React.PureComponent {
       onClick,
       onMouseMove,
       onMouseLeave,
+      horizontal,
     } = this.props;
+    if (!xScale || !yScale) return null;
+    const valueScale = horizontal ? xScale : yScale;
+    const categoryScale = horizontal ? yScale : xScale;
+    const barWidth =
+      categoryScale.barWidth || (categoryScale.bandwidth && categoryScale.bandwidth()) || 0;
+    const valueField = horizontal ? x : y;
+    const categoryField = horizontal ? y : x;
 
-    if (!xScale || !yScale || !barWidth) return null;
-
-    const maxHeight = (yScale.range() || [0])[0];
-    const offset = xScale.offset || 0;
+    const maxBarLength = (valueScale.range() || [0])[0];
```

Comment:
`Math.max(...valueScale.range())` might be a little more robust here

Reply:
Agreed!